spec/docs/logicalSource.md

```diff
@@ -11,5 +11,63 @@ A <dfn data-lt="iteration">logical iteration</dfn> is an item in the sequence pr
+There can be many different types of [=reference formulation=]. RML-Core covers CSV and JSONPath, RML-IO extends this further with XML and SQL relational databases. Other known types, and the details of how a reference formulation is handled and implemented for each data format, are specified in [[RML-IO-Registry]]. This way, implementations are not required to support all defined reference formulations to be compliant with RML-Core.
 </aside>
+
+## Reference Formulations
+
+RML-Core covers `rml:CSV` and `rml:JSONPath` as the minimum reference formulations to be supported by any implementation for referencing data in CSV and JSON documents.
+
+Each Logical Source has a reference formulation to define how to reference
+to elements of the data of the input source.
+Several reference formulations (`rml:ReferenceFormulation`)
+are defined in this specification:
+
+- `rml:CSV`: CSV documents
+- `rml:JSONPath`: JSON documents
+
+See [[RML-IO-Registry]] for a detailed specification of these reference formulations.
+
+## Access descriptions
+
+RML-Core requires the `rml:PathFile` access description to be supported by any implementation to access files with an absolute or relative path.
+This access description allows accessing files with relative and absolute paths from:
+
+- `rml:CurrentWorkingDirectory`: relative to the current working directory of the RML processor.
+- `rml:MappingDirectory`: relative to the location of the RML mapping.
+- A string Literal: a string describing an absolute path against which relative paths are resolved, similar to the Base URI in [RFC3986](https://www.rfc-editor.org/rfc/rfc3986).
+
+If `rml:root` is not specified, it defaults to `rml:CurrentWorkingDirectory`.
+
+| Property    | Domain                    | Range                                                              |
+| ----------- | ------------------------- | ------------------------------------------------------------------ |
+| `rml:root`  | `rml:FilePath`  | `rml:CurrentWorkingDirectory`, `rml:MappingDirectory` or `Literal` |
+| `rml:path`  | `rml:FilePath`  | `Literal`                                                          |
+
+Example of accessing a CSV file relative to the current working directory.
+The file's absolute path is `$CURRENT_WORKING_DIR/file.csv` where `$CURRENT_WORKING_DIR` is
+the location of the RML mapping.
+
+<pre class="ex-source">
+&lt;#RelativePathCWD&gt; a rml:LogicalSource;
+  rml:source [ a rml:FilePath;
+    rml:root rml:CurrentWorkingDirectory;
+    rml:path "./file.csv";
+  ];
+.
+</pre>
+
+Example of accessing a JSON file relative to the path of the mapping.
+The file's absolute path is `$MAPPING_DIR/file.json` where `$MAPPING_DIR` is
+the location of the RML mapping.
+
+<pre class="ex-source">
+&lt;#RelativePathMapping&gt; a rml:LogicalSource;
+  rml:source [ a rml:FilePath;
+    rml:root rml:MappingDirectory;
+    rml:path "./file.json";
+  ];
+  rml:referenceFormulation rml:JSONPath;
+  rml:iterator "$";
+.
+</pre>
```

Comment:
I'm not convinced the access description should be in core.
I think it should stay in IO, since there is where we define access descriptions

Yes, it is used in core tests, but that is not a problem IMO. It is normal to have some core module that needs other modules before it can lead to something that works. And therefor it is also normal to include some of those dependencies to run tests.

In fact I would prefer for CSV and JSON to also be introduced in IO. I think it makes the specs more easy to follow. But since reference formulation is introduced in core I'm not as strongly opposed.

Reply:
It was discussed like this during the physical meeting to also move the access description, without it, you cannot perform any test cases nor have a stand-alone RML mapping if you only support Core. Therefore, the simplest access description (next gen of `rml:source "/path/to/file.csv"`) is moved here. You mention you don't see this an issue, but having something standalone is IMO the best. Requiring people to implement some features of a module to have Core even working is not great IMO.

AFAIK, RML-Core would be the original RML actually where these things were included...

Reply:
SUGGESTION

1. Provide test cases in JSON-only with JSONPath + FIlePath
2. Add note that you can pick whatever access description and reference formulation
3. Keep JSONPath and FilePath in IO
4. XML, SQL, CSV in IO
5. ALL other reference formulations and access description in Registry

Reply:
agreed

Reply:
Force-pushed this fix, will adjust the other PRs now.